app/screens/redeem-lnurl-withdrawal-screen/redeem-bitcoin-result-screen.tsx

```diff
@@ -0,0 +1,388 @@
+              input: { walletId, amount: satAmount, memo },
+            },
+          })
+          if (errors && errors.length !== 0) {
+            console.error(errors, "error with lnInvoiceCreate")
+            setErr(LL.RedeemBitcoinScreen.error())
+            return
+          }
+
+          setInvoice(invoice)
+        } else {
+          logGeneratePaymentRequest({
+            paymentType: "lightning",
+            hasAmount: true,
+            receivingWallet: WalletCurrency.Usd,
+          })
+          const {
+            data: {
+              lnUsdInvoiceCreate: { invoice, errors },
+            },
+          } = await lnUsdInvoiceCreate({
+            variables: {
+              input: { walletId, amount: satAmountInUsd * 100, memo },
+            },
+          })
+
+          if (errors && errors.length !== 0) {
+            console.error(errors, "error with lnInvoiceCreate")
+            setErr(LL.ReceiveBitcoinScreen.error())
+            return
+          }
+          setInvoice(invoice)
+        }
+      } catch (err) {
+        console.error(err, "error with AddInvoice")
+        setErr(`${err}`)
+        throw err
+      }
+    },
+    [lnInvoiceCreate, lnUsdInvoiceCreate, LL],
+  )
+
+  const submitLNURLWithdrawRequest = async (generatedInvoice) => {
+    const url = `${callback}${callback.includes("?") ? "&" : "?"}k1=${k1}&pr=${
+      generatedInvoice.paymentRequest
+    }`
+
+    const result = await fetch(url)
+
+    if (result.ok) {
+      const lnurlResponse = await result.json()
+      if (lnurlResponse?.status?.toLowerCase() === "ok") {
+        // TODO: Set processing payment
+      } else {
+        console.error(lnurlResponse, "error with redeeming")
+        // TODO: Set failed payment
+        setErr(LL.RedeemBitcoinScreen.redeemingError())
+      }
+    } else {
+      console.error(result.text(), "error with submitting withdrawalRequest")
+      setErr(LL.RedeemBitcoinScreen.submissionError())
+    }
+  }
+
+  useEffect((): void | (() => void) => {
+    if (withdrawalInvoice) {
+      submitLNURLWithdrawRequest(withdrawalInvoice)
+    } else {
+      createWithdrawRequestInvoice({ satAmount, memo })
+    }
+  }, [walletId, withdrawalInvoice, memo, type, satAmount, createWithdrawRequestInvoice])
+
+  const renderSuccessView = useMemo(() => {
+    if (invoicePaid) {
+      return (
+        <View style={styles.container}>
+          <View {...testProps("Success Icon")} style={styles.container}>
+            <GaloyIcon name={"payment-success"} size={128} />
+          </View>
+        </View>
+      )
+    }
+    return null
+  }, [invoicePaid])
+
+  const renderStatusView = useMemo(() => {
+    if (err !== "") {
+      return (
+        <View style={styles.container}>
+          <View style={styles.errorContainer}>
+            <Text style={styles.errorText} selectable>
+              {err}
+            </Text>
+          </View>
+        </View>
+      )
+    } else if (!invoicePaid) {
```

Comment:
else is not necessary

Reply:
This actually shows the loading animation before we either get the invoice paid, or an error.

I should actually add it to it's own thing.